### ISSUE TITLE
Volume not being mounted : tls: bad certificate from web hook

### ISSUE THREAD

**Issue by snithish** [GoogleCloudPlatform/spark-on-k8s-operator#1004]
**tldr;**
Operator indeterminably fails after some time with `http: TLS handshake error from 10.0.0.131:56084: remote error: tls: bad certificate`. Killing the operator fixes this issue.

**Details**
We have Spark application mounting an NFS volume in the driver, and when launched by the SparkOperator, sometimes the volume does not get mounted. The application gets submitted but the volume mount is missing, when this happens in the SparkOperator log show 
`2020/08/25 11:25:21 http: TLS handshake error from 10.0.0.131:56084: remote error: tls: bad certificate` leading me to believe TLS verification with web hook fails.

**Comment by FloraZhang** [comment]
Hi repo master,

I'm also seeing this issue, with sparkoperator log as below:

```
[root@node1 opt]# kubectl logs spark-operator-5b84f9db54-p2jhp

I1028 14:37:07.902848       9 controller.go:773] Update the status of SparkApplication default/rta-windower-app from:
{
  "sparkApplicationId": "spark-6db39d5a808243d19839958f924eac59",
  "submissionID": "64af25c7-0371-4b43-95eb-25a97b6d83ee",
  "lastSubmissionAttemptTime": "2020-10-28T14:36:21Z",
  "terminationTime": "2020-10-28T14:36:35Z",
  "driverInfo": {
    "webUIServiceName": "rta-windower-app-ui-svc",
    "webUIPort": 40450,
    "webUIAddress": "10.233.50.95:40450",
    "podName": "rta-windower-app-driver"
  },
  "applicationState": {
    "state": "FAILING",
    "errorMessage": "driver container failed with ExitCode: 134, Reason: Error"
  },
  "executionAttempts": 13,
  "submissionAttempts": 1
}
to:
{
  "sparkApplicationId": "spark-6db39d5a808243d19839958f924eac59",
  "submissionID": "64af25c7-0371-4b43-95eb-25a97b6d83ee",
  "lastSubmissionAttemptTime": "2020-10-28T14:36:21Z",
  "terminationTime": "2020-10-28T14:36:35Z",
  "driverInfo": {
    "webUIServiceName": "rta-windower-app-ui-svc",
    "webUIPort": 40450,
    "webUIAddress": "10.233.50.95:40450",
    "podName": "rta-windower-app-driver"
  },
  "applicationState": {
    "state": "PENDING_RERUN",
    "errorMessage": "driver container failed with ExitCode: 134, Reason: Error"
  },
  "executionAttempts": 13,
  "submissionAttempts": 1
}
I1028 14:37:07.937329       9 controller.go:265] Ending processing key: "default/rta-windower-app"
I1028 14:37:07.937380       9 controller.go:258] Starting processing key: "default/rta-windower-app"
I1028 14:37:07.937465       9 controller.go:569] SparkApplication default/rta-windower-app is pending rerun
I1028 14:37:07.943499       9 controller.go:571] Resources for SparkApplication default/rta-windower-app successfully deleted
I1028 14:37:07.943556       9 monitoring_config.go:49] Creating a ConfigMap for metrics and Prometheus configurations.
I1028 14:37:07.943712       9 event.go:274] Event(v1.ObjectReference{Kind:"SparkApplication", Namespace:"default", Name:"rta-windower-app", UID:"ebf2b582-77d9-460b-bfb5-dcc52431f673", APIVersion:"sparkoperator.k8s.io/v1beta2", ResourceVersion:"659784", FieldPath:""}): type: 'Warning' reason: 'SparkApplicationPendingRerun' SparkApplication rta-windower-app is pending rerun
I1028 14:37:07.958926       9 submission.go:65] spark-submit arguments: [/opt/spark/bin/spark-submit --class com.vex.rta.spark.windower.WindowerApp --master k8s://https://10.233.0.1:443 --deploy-mode cluster --conf spark.kubernetes.namespace=default --conf spark.app.name=rta-windower-app --conf spark.kubernetes.driver.pod.name=rta-windower-app-driver --jars local:///opt/vex/spark/extra-jars/spark-sql-kafka-0-10_2.11-2.4.4.jar,local:///opt/vex/spark/extra-jars/spark-streaming-kafka-0-10_2.11-2.4.4.jar,local:///opt/vex/spark/extra-jars/kafka-clients-2.0.0.jar,local:///opt/vex/spark/extra-jars/jmx_prometheus_javaagent-0.12.0.jar,local:///opt/vex/spark/extra-jars/postgresql-42.1.4.jar,local:///opt/vex/spark/jars/vertica-jdbc-9.3.0-0.jar --files local:///opt/vex/rta/spark-app/application.conf --conf spark.kubernetes.container.image=registry.vex.local/analytics-rtanalytics-baseline-app:20.11.0-beta.0 --conf spark.kubernetes.container.image.pullPolicy=IfNotPresent --conf spark.kubernetes.submission.waitAppCompletion=false --conf spark.memory.storageFraction=0.1 --conf spark.memory.fraction=0.85 --conf spark.ui.port=40450 --conf spark.kryoserializer.buffer.max=32m --conf spark.serializer=org.apache.spark.serializer.KryoSerializer --conf spark.sql.streaming.checkpointLocation=file:///opt/checkpoint-data --conf spark.cleaner.periodicGC.interval=30s --conf spark.kryo.unsafe=true --conf spark.metrics.namespace=default.rta-windower-app --conf spark.metrics.conf=/etc/metrics/conf/metrics.properties --conf spark.kubernetes.driver.memoryOverhead=0.4 --conf spark.kubernetes.executor.memoryOverhead=0.4 --conf spark.kubernetes.driver.label.sparkoperator.k8s.io/app-name=rta-windower-app --conf spark.kubernetes.driver.label.sparkoperator.k8s.io/launched-by-spark-operator=true --conf spark.kubernetes.driver.label.sparkoperator.k8s.io/submission-id=3898ae58-9a71-469d-a7d5-f304e1ecca31 --conf spark.driver.cores=1 --conf spark.kubernetes.driver.limit.cores=1000m --conf spark.driver.memory=1G --conf spark.driver.memoryOverhead=1G --conf spark.kubernetes.authenticate.driver.serviceAccountName=rtanalytics --conf spark.driver.extraJavaOptions=-Dlog4j.configuration=file:///opt/vex/spark/conf/windower_driver-log4j.properties -Dvex.application.conf.dir=/opt/vex/rta/spark-app -XX:+PrintClassHistogramBeforeFullGC  -verbose:gc -XX:+PrintGCDetails -XX:+PrintGCTimeStamps -XX:+PrintGCDateStamps -XX:+PrintPromotionFailure -XX:+PrintAdaptiveSizePolicy -XX:+PrintReferenceGC -Xloggc:/opt/spark/logs/nsposgc.log -XX:+UseGCLogFileRotation -XX:NumberOfGCLogFiles=20 -XX:GCLogFileSize=5M -XX:+UseCompressedOops -XX:+UseG1GC -XX:+UseContainerSupport -XX:InitialRAMPercentage=50.0 -XX:MinRAMPercentage=20.0 -XX:MaxRAMPercentage=85.0  -XX:InitiatingHeapOccupancyPercent=25 -javaagent:/opt/vex/spark/extra-jars/jmx_prometheus_javaagent-0.12.0.jar=8090:/etc/metrics/conf/prometheus.yaml --conf spark.kubernetes.driver.label.helm.sh/chart=analytics-rtanalytics-baseline-app-20.11.0-beta.0 --conf spark.kubernetes.driver.label.app.kubernetes.io/instance=rta-baseline-app --conf spark.kubernetes.driver.label.app.kubernetes.io/managed-by=Tiller --conf spark.kubernetes.driver.label.app.kubernetes.io/name=analytics-rtanalytics-baseline-app --conf spark.kubernetes.driver.label.sparkapp=windower --conf spark.kubernetes.driver.annotation.prometheus.io/port=8090 --conf spark.kubernetes.driver.annotation.prometheus.io/path=/metrics --conf spark.kubernetes.driver.annotation.prometheus.io/scrape=true --conf spark.kubernetes.executor.label.sparkoperator.k8s.io/app-name=rta-windower-app --conf spark.kubernetes.executor.label.sparkoperator.k8s.io/launched-by-spark-operator=true --conf spark.kubernetes.executor.label.sparkoperator.k8s.io/submission-id=3898ae58-9a71-469d-a7d5-f304e1ecca31 --conf spark.executor.instances=1 --conf spark.executor.cores=1 --conf spark.kubernetes.executor.limit.cores=1000m --conf spark.executor.memory=1G --conf spark.executor.memoryOverhead=3G --conf spark.kubernetes.executor.label.app.kubernetes.io/managed-by=Tiller --conf spark.kubernetes.executor.label.app.kubernetes.io/name=analytics-rtanalytics-baseline-app --conf spark.kubernetes.executor.label.helm.sh/chart=analytics-rtanalytics-baseline-app-20.11.0-beta.0 --conf spark.kubernetes.executor.label.app.kubernetes.io/instance=rta-baseline-app --conf spark.kubernetes.executor.annotation.prometheus.io/scrape=true --conf spark.kubernetes.executor.annotation.prometheus.io/port=8090 --conf spark.kubernetes.executor.annotation.prometheus.io/path=/metrics --conf spark.executor.extraJavaOptions=-Dlog4j.configuration=file:///opt/vex/spark/conf/windower_executor-log4j.properties -Dvex.application.conf.dir=/opt/vex/rta/spark-app -XX:+PrintClassHistogramBeforeFullGC  -verbose:gc -XX:+PrintGCDetails -XX:+PrintGCTimeStamps -XX:+PrintGCDateStamps -XX:+PrintPromotionFailure -XX:+PrintAdaptiveSizePolicy -XX:+PrintReferenceGC -Xloggc:/opt/spark/logs/nsposgc.log -XX:+UseGCLogFileRotation -XX:NumberOfGCLogFiles=20 -XX:GCLogFileSize=5M -XX:+UseCompressedOops -XX:+UseG1GC -XX:+UseContainerSupport -XX:InitialRAMPercentage=50.0 -XX:MinRAMPercentage=20.0 -XX:MaxRAMPercentage=85.0  -XX:InitiatingHeapOccupancyPercent=25 -javaagent:/opt/vex/spark/extra-jars/jmx_prometheus_javaagent-0.12.0.jar=8090:/etc/metrics/conf/prometheus.yaml local:///opt/vex/rta/spark-app/analytics-rtanalytics-baseline-spark-all.jar]
2020/10/28 14:37:19 http: TLS handshake error from 10.233.90.0:53746: remote error: tls: bad certificate
```

When driver pod is created, in the pod log I'm seeing below error. I think it's due to the operator failed to mount certain configmap to driver pod.
```
[root@node1 ~]# kubectl logs rta-windower-app-driver
Exception in thread "main" java.lang.reflect.InvocationTargetException
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at sun.instrument.InstrumentationImpl.loadClassAndStartAgent(InstrumentationImpl.java:386)
        at sun.instrument.InstrumentationImpl.loadClassAndCallPremain(InstrumentationImpl.java:401)
Caused by: java.io.FileNotFoundException: /etc/metrics/conf/prometheus.yaml (No such file or directory)
        at java.io.FileInputStream.open0(Native Method)
        at java.io.FileInputStream.open(FileInputStream.java:195)
        at java.io.FileInputStream.<init>(FileInputStream.java:138)
        at java.io.FileReader.<init>(FileReader.java:72)
        at io.prometheus.jmx.shaded.io.prometheus.jmx.JmxCollector.<init>(JmxCollector.java:74)
FATAL ERROR in native method: processing of -javaagent failed
        at io.prometheus.jmx.shaded.io.prometheus.jmx.JavaAgent.premain(JavaAgent.java:29)
        ... 6 more
```

**Comment by fisache** [comment]
hi, it looks like same issue https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/pull/1058.

**Comment by kudr2001** [comment]
@snithish @FloraZhang It looks like the bug is fixed in "master" branch by @fisache. I also encountered `http: TLS handshake error from 10.0.0.131:56084: remote error: tls: bad certificate` trying to use `hadoopConfigMap`. But then I reinstalled spark-operator from "master" branch using `--set image.tag=latest` parameter and the bug is dissappeared.

**Comment by snithish** [comment]
Thanks for the insight @kudr2001

**Comment by birnemeier** [comment]
I'm sorry, but I cannot confirm that using `latest` fixes the issue (hadoopConfigMap not being mounted). 

API-Server shows:
```
Failed calling webhook, failing open webhook.sparkoperator.k8s.io: failed calling webhook "webhook.sparkoperator.k8s.io": Post "https://spark-operator-webhook.spark-operator.svc:443/webhook?timeout=30s": x509: certificate relies on legacy Common Name field, use SANs or temporarily enable Common Name matching with GODEBUG=x509ignoreCN=0
```

spark-operator log shows:
```
http: TLS handshake error from 10.233.69.0:6340: remote error: tls: bad certificate
```

**Comment by andreaTP** [comment]
Setting `operatorVersion: latest` and using the helm chart works for me. Does anyone have a hint on the commit that fixed the issue?

**Comment by jgoeres** [comment]
We ran into the same issue - after initial deployment, Spark jobs work fine. After a while, driver pods are being started that are lacking tolerations and volume mounts. Operator logs show the "bad certificate" error.
Can someone shed some light on the exact failure mechanism that is at play here? 
If the patch (fixed CN for the generated webhook certificate) mentioned in the referenced issue is indeed fixing this - why does it work initially, and does not fail right from the start?

**Comment by jgoeres** [comment]
Meanwhile, I backported the change done for 1.2.2 to the 1.1.0 we are on. Same effect. After a while, the webhook fails with the bad certificate error. 
I want to add that we moved from EKS 1.15 to 1.18, maybe this is related?

I can also see a number of rather new issues being reported that all seem to point into the same direction:

https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/issues/1168
https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/issues/1161

**Comment by weicheng113** [comment]
> Hi repo master,
> 
> I'm also seeing this issue, with sparkoperator log as below:
> 
> ```
> [root@node1 opt]# kubectl logs spark-operator-5b84f9db54-p2jhp
> 
> I1028 14:37:07.902848       9 controller.go:773] Update the status of SparkApplication default/rta-windower-app from:
> {
>   "sparkApplicationId": "spark-6db39d5a808243d19839958f924eac59",
>   "submissionID": "64af25c7-0371-4b43-95eb-25a97b6d83ee",
>   "lastSubmissionAttemptTime": "2020-10-28T14:36:21Z",
>   "terminationTime": "2020-10-28T14:36:35Z",
>   "driverInfo": {
>     "webUIServiceName": "rta-windower-app-ui-svc",
>     "webUIPort": 40450,
>     "webUIAddress": "10.233.50.95:40450",
>     "podName": "rta-windower-app-driver"
>   },
>   "applicationState": {
>     "state": "FAILING",
>     "errorMessage": "driver container failed with ExitCode: 134, Reason: Error"
>   },
>   "executionAttempts": 13,
>   "submissionAttempts": 1
> }
> to:
> {
>   "sparkApplicationId": "spark-6db39d5a808243d19839958f924eac59",
>   "submissionID": "64af25c7-0371-4b43-95eb-25a97b6d83ee",
>   "lastSubmissionAttemptTime": "2020-10-28T14:36:21Z",
>   "terminationTime": "2020-10-28T14:36:35Z",
>   "driverInfo": {
>     "webUIServiceName": "rta-windower-app-ui-svc",
>     "webUIPort": 40450,
>     "webUIAddress": "10.233.50.95:40450",
>     "podName": "rta-windower-app-driver"
>   },
>   "applicationState": {
>     "state": "PENDING_RERUN",
>     "errorMessage": "driver container failed with ExitCode: 134, Reason: Error"
>   },
>   "executionAttempts": 13,
>   "submissionAttempts": 1
> }
> I1028 14:37:07.937329       9 controller.go:265] Ending processing key: "default/rta-windower-app"
> I1028 14:37:07.937380       9 controller.go:258] Starting processing key: "default/rta-windower-app"
> I1028 14:37:07.937465       9 controller.go:569] SparkApplication default/rta-windower-app is pending rerun
> I1028 14:37:07.943499       9 controller.go:571] Resources for SparkApplication default/rta-windower-app successfully deleted
> I1028 14:37:07.943556       9 monitoring_config.go:49] Creating a ConfigMap for metrics and Prometheus configurations.
> I1028 14:37:07.943712       9 event.go:274] Event(v1.ObjectReference{Kind:"SparkApplication", Namespace:"default", Name:"rta-windower-app", UID:"ebf2b582-77d9-460b-bfb5-dcc52431f673", APIVersion:"sparkoperator.k8s.io/v1beta2", ResourceVersion:"659784", FieldPath:""}): type: 'Warning' reason: 'SparkApplicationPendingRerun' SparkApplication rta-windower-app is pending rerun
> I1028 14:37:07.958926       9 submission.go:65] spark-submit arguments: [/opt/spark/bin/spark-submit --class com.vex.rta.spark.windower.WindowerApp --master k8s://https://10.233.0.1:443 --deploy-mode cluster --conf spark.kubernetes.namespace=default --conf spark.app.name=rta-windower-app --conf spark.kubernetes.driver.pod.name=rta-windower-app-driver --jars local:///opt/vex/spark/extra-jars/spark-sql-kafka-0-10_2.11-2.4.4.jar,local:///opt/vex/spark/extra-jars/spark-streaming-kafka-0-10_2.11-2.4.4.jar,local:///opt/vex/spark/extra-jars/kafka-clients-2.0.0.jar,local:///opt/vex/spark/extra-jars/jmx_prometheus_javaagent-0.12.0.jar,local:///opt/vex/spark/extra-jars/postgresql-42.1.4.jar,local:///opt/vex/spark/jars/vertica-jdbc-9.3.0-0.jar --files local:///opt/vex/rta/spark-app/application.conf --conf spark.kubernetes.container.image=registry.vex.local/analytics-rtanalytics-baseline-app:20.11.0-beta.0 --conf spark.kubernetes.container.image.pullPolicy=IfNotPresent --conf spark.kubernetes.submission.waitAppCompletion=false --conf spark.memory.storageFraction=0.1 --conf spark.memory.fraction=0.85 --conf spark.ui.port=40450 --conf spark.kryoserializer.buffer.max=32m --conf spark.serializer=org.apache.spark.serializer.KryoSerializer --conf spark.sql.streaming.checkpointLocation=file:///opt/checkpoint-data --conf spark.cleaner.periodicGC.interval=30s --conf spark.kryo.unsafe=true --conf spark.metrics.namespace=default.rta-windower-app --conf spark.metrics.conf=/etc/metrics/conf/metrics.properties --conf spark.kubernetes.driver.memoryOverhead=0.4 --conf spark.kubernetes.executor.memoryOverhead=0.4 --conf spark.kubernetes.driver.label.sparkoperator.k8s.io/app-name=rta-windower-app --conf spark.kubernetes.driver.label.sparkoperator.k8s.io/launched-by-spark-operator=true --conf spark.kubernetes.driver.label.sparkoperator.k8s.io/submission-id=3898ae58-9a71-469d-a7d5-f304e1ecca31 --conf spark.driver.cores=1 --conf spark.kubernetes.driver.limit.cores=1000m --conf spark.driver.memory=1G --conf spark.driver.memoryOverhead=1G --conf spark.kubernetes.authenticate.driver.serviceAccountName=rtanalytics --conf spark.driver.extraJavaOptions=-Dlog4j.configuration=file:///opt/vex/spark/conf/windower_driver-log4j.properties -Dvex.application.conf.dir=/opt/vex/rta/spark-app -XX:+PrintClassHistogramBeforeFullGC  -verbose:gc -XX:+PrintGCDetails -XX:+PrintGCTimeStamps -XX:+PrintGCDateStamps -XX:+PrintPromotionFailure -XX:+PrintAdaptiveSizePolicy -XX:+PrintReferenceGC -Xloggc:/opt/spark/logs/nsposgc.log -XX:+UseGCLogFileRotation -XX:NumberOfGCLogFiles=20 -XX:GCLogFileSize=5M -XX:+UseCompressedOops -XX:+UseG1GC -XX:+UseContainerSupport -XX:InitialRAMPercentage=50.0 -XX:MinRAMPercentage=20.0 -XX:MaxRAMPercentage=85.0  -XX:InitiatingHeapOccupancyPercent=25 -javaagent:/opt/vex/spark/extra-jars/jmx_prometheus_javaagent-0.12.0.jar=8090:/etc/metrics/conf/prometheus.yaml --conf spark.kubernetes.driver.label.helm.sh/chart=analytics-rtanalytics-baseline-app-20.11.0-beta.0 --conf spark.kubernetes.driver.label.app.kubernetes.io/instance=rta-baseline-app --conf spark.kubernetes.driver.label.app.kubernetes.io/managed-by=Tiller --conf spark.kubernetes.driver.label.app.kubernetes.io/name=analytics-rtanalytics-baseline-app --conf spark.kubernetes.driver.label.sparkapp=windower --conf spark.kubernetes.driver.annotation.prometheus.io/port=8090 --conf spark.kubernetes.driver.annotation.prometheus.io/path=/metrics --conf spark.kubernetes.driver.annotation.prometheus.io/scrape=true --conf spark.kubernetes.executor.label.sparkoperator.k8s.io/app-name=rta-windower-app --conf spark.kubernetes.executor.label.sparkoperator.k8s.io/launched-by-spark-operator=true --conf spark.kubernetes.executor.label.sparkoperator.k8s.io/submission-id=3898ae58-9a71-469d-a7d5-f304e1ecca31 --conf spark.executor.instances=1 --conf spark.executor.cores=1 --conf spark.kubernetes.executor.limit.cores=1000m --conf spark.executor.memory=1G --conf spark.executor.memoryOverhead=3G --conf spark.kubernetes.executor.label.app.kubernetes.io/managed-by=Tiller --conf spark.kubernetes.executor.label.app.kubernetes.io/name=analytics-rtanalytics-baseline-app --conf spark.kubernetes.executor.label.helm.sh/chart=analytics-rtanalytics-baseline-app-20.11.0-beta.0 --conf spark.kubernetes.executor.label.app.kubernetes.io/instance=rta-baseline-app --conf spark.kubernetes.executor.annotation.prometheus.io/scrape=true --conf spark.kubernetes.executor.annotation.prometheus.io/port=8090 --conf spark.kubernetes.executor.annotation.prometheus.io/path=/metrics --conf spark.executor.extraJavaOptions=-Dlog4j.configuration=file:///opt/vex/spark/conf/windower_executor-log4j.properties -Dvex.application.conf.dir=/opt/vex/rta/spark-app -XX:+PrintClassHistogramBeforeFullGC  -verbose:gc -XX:+PrintGCDetails -XX:+PrintGCTimeStamps -XX:+PrintGCDateStamps -XX:+PrintPromotionFailure -XX:+PrintAdaptiveSizePolicy -XX:+PrintReferenceGC -Xloggc:/opt/spark/logs/nsposgc.log -XX:+UseGCLogFileRotation -XX:NumberOfGCLogFiles=20 -XX:GCLogFileSize=5M -XX:+UseCompressedOops -XX:+UseG1GC -XX:+UseContainerSupport -XX:InitialRAMPercentage=50.0 -XX:MinRAMPercentage=20.0 -XX:MaxRAMPercentage=85.0  -XX:InitiatingHeapOccupancyPercent=25 -javaagent:/opt/vex/spark/extra-jars/jmx_prometheus_javaagent-0.12.0.jar=8090:/etc/metrics/conf/prometheus.yaml local:///opt/vex/rta/spark-app/analytics-rtanalytics-baseline-spark-all.jar]
> 2020/10/28 14:37:19 http: TLS handshake error from 10.233.90.0:53746: remote error: tls: bad certificate
> ```
> 
> When driver pod is created, in the pod log I'm seeing below error. I think it's due to the operator failed to mount certain configmap to driver pod.
> 
> ```
> [root@node1 ~]# kubectl logs rta-windower-app-driver
> Exception in thread "main" java.lang.reflect.InvocationTargetException
>         at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
>         at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
>         at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
>         at java.lang.reflect.Method.invoke(Method.java:498)
>         at sun.instrument.InstrumentationImpl.loadClassAndStartAgent(InstrumentationImpl.java:386)
>         at sun.instrument.InstrumentationImpl.loadClassAndCallPremain(InstrumentationImpl.java:401)
> Caused by: java.io.FileNotFoundException: /etc/metrics/conf/prometheus.yaml (No such file or directory)
>         at java.io.FileInputStream.open0(Native Method)
>         at java.io.FileInputStream.open(FileInputStream.java:195)
>         at java.io.FileInputStream.<init>(FileInputStream.java:138)
>         at java.io.FileReader.<init>(FileReader.java:72)
>         at io.prometheus.jmx.shaded.io.prometheus.jmx.JmxCollector.<init>(JmxCollector.java:74)
> FATAL ERROR in native method: processing of -javaagent failed
>         at io.prometheus.jmx.shaded.io.prometheus.jmx.JavaAgent.premain(JavaAgent.java:29)
>         ... 6 more
> ```

We are having the same issue and get the exact exception message in #1168 .

**Comment by jacobhjkim** [comment]
Is this solved by https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/pull/1058?

**Comment by snithish** [comment]
Closing this issue, as I did not observe this behavior in newer versions.

**Comment by domenicbove** [comment]
Hi I am observing this error, and am a little confused. This PR: https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/pull/1058 looks like it was merged into master. But I still see the invalid line here: https://github.com/nicholas-fwang/spark-on-k8s-operator/blob/master/hack/gencerts.sh#L108

@snithish what version are you running?
